src/utils/resolver-pipeline/testing/4_modified/modify.xspec

```diff
@@ -1158,7 +1158,7 @@
         </x:scenario>
         <x:scenario label="Edge case: Empty $modifications param">
             <x:call function="oscal:patches-to-id-targeting-ancestor">
-                <x:param name="here" select="id('a1-stmt')">
+                <x:param name="here" select="//*[@id='a1-stmt']">
```

Comment:
For the record: `id(...)` in this context was incorrect all along, so I wondered why the test passed in Oxygen but not on the command line. I believe it's a difference in the Saxon edition in use. Oxygen uses Saxon-EE in the default XSpec transformation scenario, but on the command line I was using Saxon-HE. When I force Oxygen to use Saxon-HE, I get the same error as in the command line run.

I speculate that Saxon-EE might have some optimization that makes the function call bypass data type checking of the first input argument. Perhaps Saxon-EE figures out that in the function, the `$alterations` variable is empty, so the return value of the function is empty regardless of the first input.

Reply:
The CI/CD pipeline uses Saxon-HE 11.5. The latest version is 12.4, but @nikitawootten-nist indicated earlier attempts to upgrade to Saxon-HE 12.0, caused issues. I haven't tried the latest version. Which version of Saxon-HE have you used?

Reply:
In my comment above, I used Saxon-EE 12.3 with Oxygen and Saxon-HE 12.3 outside Oxygen.

Just now, I tried the 6 XSpec files in this pull request with Saxon-HE 11.5 outside Oxygen (in a Windows shell using `xspec.bat`). Still no failures.

Reply:
Thank you very much, @galtm ! I greatly appreciate your help, Amanda. I will run tomorrow all tests.

Reply:
On the Checks tab of this PR, in the "Run all pipelines" section, I see evidence of these tests running (e.g., log includes "PENDING: (Needs confirmation about latest expected result) ..include-controls with parent controls (default choice)"), and I do not see failures. Line 1650  of that log says, "All test suites passed ✅"

Reply:
Functionality of `id()` can depend on a schema binding, which is one of the features offered by Saxon-PE/EE, available in oXygen but not in Saxon-HE. So I am not entirely surprised to see that wiggle.